src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php

```diff
@@ -288,4 +288,21 @@ public function unescapeValue($value)
 
         return $value;
     }
+
+    /**
+     * @internal to be removed in 4.0
+     */
+    public static function normalizeName($name)
+    {
+        if (0 === strpos($name, 'env(') && ')' === substr($name, -1) && 'env()' !== $name) {
+            return strtolower($name);
+        }
+
+        $normalizedName = strtolower($name);
+        if ($normalizedName !== (string) $name) {
+            @trigger_error(sprintf('Parameter names will be made case sensitive in Symfony 4.0. Using "%s" instead of "%s" is deprecated since version 3.4.', $name, $normalizedName), E_USER_DEPRECATED);
```

Comment:
That's what was done for services name but the complexity to do it is not worth it for parameters imo. If you don't want to have these deprecations, you'll still be able to quickly upgrade to 4.0.